app/controllers/thredded/topics_controller.rb

```diff
@@ -3,7 +3,7 @@ module Thredded
   class TopicsController < Thredded::ApplicationController
     before_action :thredded_require_login!,
                   only: [:edit, :new, :update, :create, :destroy]
-    before_action :update_user_activity
+    after_action :update_user_activity
```

Comment:
ha. this should have been this way the entire time, now thatI look at it. Wouldn't be surprised if this was a big part of the read-tracking bug